pkg/generic/thrift/json_go116plus_amd64.go

```diff
@@ -0,0 +1,111 @@
+//go:build amd64 && go1.16
+// +build amd64,go1.16
+
+/*
+ * Copyright 2021 CloudWeGo Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package thrift
+
+import (
+	"context"
+
+	"github.com/apache/thrift/lib/go/thrift"
+	"github.com/bytedance/gopkg/lang/mcache"
+	"github.com/cloudwego/dynamicgo/conv/j2t"
+
+	"github.com/cloudwego/kitex/pkg/generic/descriptor"
+	"github.com/cloudwego/kitex/pkg/protocol/bthrift"
+	"github.com/cloudwego/kitex/pkg/remote/codec/perrors"
+	cthrift "github.com/cloudwego/kitex/pkg/remote/codec/thrift"
+	"github.com/cloudwego/kitex/pkg/utils"
+)
+
+// Write write json string to out thrift.TProtocol
+func (m *WriteJSON) Write(ctx context.Context, out thrift.TProtocol, msg interface{}, requestBase *Base) error {
+	if m.enableDynamicgo {
+		var msgLen int
+		_, isVoid := msg.(descriptor.Void)
+		transBuff, isString := msg.(string)
+		if !isVoid && !isString {
+			return perrors.NewProtocolErrorWithType(perrors.InvalidData, "decode msg failed, is neither void nor string")
+		} else if isVoid {
+			msgLen = 0
+		} else {
+			msgLen = len(transBuff)
+		}
+
+		if m.dty == nil {
+			return perrors.NewProtocolErrorWithMsg("svcDsc.DynamicgoDsc is nil")
```

Comment:
This judgement should be done when constructing WriteJSON

Reply:
@jayantxie I think it is ok here because we need dty only under amd64 && go1.16 (this file), what do you think?

Reply:
You can also split the initialization logic for constructing operations such as building `DescriptorProvider` with build tags, so that descriptor of Dynamicgo can be used only in required scenarios

Reply:
fixed